src/fetcher.ts

```diff
@@ -0,0 +1,34 @@
+import { Pair } from './entities/pair'
+import invariant from 'tiny-invariant'
+import ERC20 from './abis/hrc20_abi.json'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+
+function decimalize(amount: number, decimals: number): JSBI {
+    return JSBI.multiply(JSBI.BigInt(amount), JSBI.exponentiate(JSBI.BigInt(10), JSBI.BigInt(decimals)))
+  }
+
+/**
+ * Contains methods for constructing instances of pairs and tokens from on-chain data.
+ */
+export abstract class Fetcher {
+  /**
+   * Cannot be constructed.
+   */
+  private constructor() {}
+
+  /**
+   * Fetches information about a pair and constructs a pair from the given two tokens.
+   * @param tokenA first token
+   * @param tokenB second token
+   * @param provider the provider to use to fetch the data
+   */
+  public static async fetchPairData(
+    tokenA: Token,
+    tokenB: Token,
+    factoryAddress: string,
+  ): Promise<Pair> {
+    invariant(tokenA.chainId === tokenB.chainId, 'CHAIN_ID')
+    return new Pair(CurrencyAmount.fromRawAmount(tokenA, decimalize(1, tokenA.decimals)), CurrencyAmount.fromRawAmount(tokenB, decimalize(1, tokenB.decimals)), factoryAddress)

```

Comment:
I'm pretty sure the pair can't just have 1 as the raw amount. I think it should be the token balances of the DEX pair

Reply:
you are correct